### ISSUE TITLE
How does development in DartPad work today?

### ISSUE THREAD

**Issue by kasperpeulen** [dart-lang/dart-pad#714]
I'm not able to get development in DartPad to work in Dartium.
If I just do `pub get` and then `pub serve`. I get an error that codemirror is not found.

After doing `grind build`, everything in the build folder seems to work correctly in chrome, but that is not really ideal.

After doing grind build, the codemirror problems seems to be resolved for development, but still not everything works:

![image](https://cloud.githubusercontent.com/assets/1035299/12756682/19699746-c9d5-11e5-9398-26821278411f.png)

I get this error. Interestingly, I don't get this error with `pub serve` in Chrome. Is this a Dartium issue?

**Comment by lukechurch** [comment]
Hey Kasper - good to see you again :)

That error actually looks like it's coming back from dart-services as a compile error which is very strange indeed. Could you try it again and if you see anything in the dev console post it here?

I have generally just been doing grind builds, which I agree is a little tiresome @devoncarew may have a better answer.

**Comment by kasperpeulen** [comment]
@lukechurch The console is not giving any errors:

![image](https://cloud.githubusercontent.com/assets/1035299/12757703/6b4044a8-c9d9-11e5-96e5-eaa3cf69a7bb.png)

> I have generally just been doing grind builds

Yes, so `pub serve` using chrome also works, which is must faster, but also not ideal for development.

**Comment by devoncarew** [comment]
So my workflow has been:

- grind build, and test the production build in build/web/ w/ chrome
- start a local server (not pub serve), and test w/ Dartium in web/. I have been using `mserve`, but anything simple will work, like `python -m SimpleHTTPServer`

It might work to develop w/ pub serve, but the project really hasn't been designed w/ it in mind. 

I also use the `bin/services.dart` entry-point in dart-services to develop locally, which doesn't require appengine.

**Comment by kasperpeulen** [comment]
I get the same error message with both `python -m SimpleHTTPServer` and with `mserve`. Also if I run `bin/services.dart` locally. 

So you are not getting any errors in Dartium?

**Comment by kasperpeulen** [comment]
@devoncarew I was wondering, are you able to reproduce the same errors I'm getting, or is this maybe some problem with my setup?

**Comment by devoncarew** [comment]
Will check shortly :)

**Comment by devoncarew** [comment]
Yeah, so I verified that things are working for me. Some more details:

- edit lib/services/common.dart to use the `http://127.0.0.1:8080/` url
- in `dart-pad/`, up-to-date w/ master, run pub get and then grind build
- run `mserve`
- in parallel, from `dart-services/`, run dart bin/services.dart
- open dartium, browser to localhost:8000 (the mserve port)
- browse to web/index.html, or build/web/index.html
- verified that the site loaded, that the traffic went between the local dart-services instance and the browser, and things like compiling and running worked

Not sure where things are going off the rails for you. If you watch the traffic in devtools, is that error displayed coming from the local dart-services server, or happening in the browser before or after the request?

`Dart VM version: 1.15.0-dev.2.0 (Wed Feb 10 02:48:16 2016) on "macos_x64"`
